### PR TITLE
feat(api): add read-only /v1/containers endpoint

### PR DESCRIPTION
**Pull request by barnabasbusa**
## Summary

Adds an optional, read-only `GET /v1/containers` HTTP API endpoint (enabled with `--http-api-containers` / `WATCHTOWER_HTTP_API_CONTAINERS`) that lists each watched container's current image identity:

- `name` — container name
- `image` — image reference with tag
- `image_id` — local image config ID
- `running_digest` — the registry manifest digest the running image was pulled from (from the image's `RepoDigests`), directly comparable to a registry's `Docker-Content-Digest`. Empty for locally-built images with no registry reference.

It's the read-only counterpart to `/v1/update`: it lets an external orchestrator see what each container is actually running and compare it against a registry **without pulling any image layers**. Addresses the use case in #979.

## Motivation

We drive gated, sequential image rollouts across an Ethereum node fleet and need to know which container is on which digest — both to decide what to roll and to confirm a node recovered onto the new digest afterwards. Watchtower already inspects this information internally; this just exposes it read-only.

## Changes

- New `pkg/api/containers` handler (+ test)
- Wire-up in `internal/api` behind the new flag, listing via the existing container client + filter and reading `ImageInfo().RepoDigests`
- Flag + config plumbing (`internal/flags`, `pkg/types`, `cmd/root.go`)
- Docs: HTTP API + arguments pages

## Notes

- Token-authenticated and GET-only, same as `/v1/metrics`.
- Coexists with `--http-api-update` and `--http-api-metrics`.

## Test plan

- [x] `go build ./...`
- [x] `go test ./pkg/api/containers/...`
- [ ] Manual: `curl -H "Authorization: Bearer <token>" localhost:8080/v1/containers`

## Summary by CodeRabbit

* **New Features**
  * Added a read-only GET /v1/containers HTTP API that lists watched containers with name, image, image ID, and registry-derived digest; response includes containers, count, timestamp, and api_version.
  * Added a CLI flag (--http-api-containers) and environment variable to enable the containers API.

* **Documentation**
  * Documented the new endpoint, response fields, and how to enable the option.

* **Tests**
  * Added unit and integration tests covering success, empty results, auth failures, handler errors, and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/nicholas-fedor/watchtower/pull/1700?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: docs/advanced-features/http-api/index.md [hunk 1]
```diff
@@ -9,10 +9,11 @@ Watchtower has an [optional](../../configuration/arguments/index.md#http_api_mod
...
-|     [Update](#http_api_update)     |   `POST`   | `/v1/update`  | [`image`](#image_parameter_usage), [`async`](#asynchronous_updates) | Triggers container updates and returns JSON results of the operation |
-| [Metrics](../metrics-api/index.md) |   `GET`    | `/v1/metrics` |                                                                     |  Exposes Prometheus-compatible metrics for monitoring and alerting   |
+|              **Name**              | **Method** |   **Endpoint**   |                           **Parameters**                            |                           **Description**                            |
+|:----------------------------------:|:----------:|:----------------:|:-------------------------------------------------------------------:|:--------------------------------------------------------------------:|
+|     [Update](#http_api_update)     |   `POST`   |   `/v1/update`   | [`image`](#image_parameter_usage), [`async`](#asynchronous_updates) | Triggers container updates and returns JSON results of the operation |
```
[the remark this answers is not in the document]
Comment (nicholas-fedor) [reply docs/advanced-features/http-api/index.md:14]: This is a a false positive as the configuration is a quirk of the current Mkdocs config.

**Comment by nicholas-fedor** [comment]
Thank you for submitting the feature request / PR.
I have reviewed and made some modifications, including opting for the `digest` field, rather than `running_digest`. 

Testing shows functionality to be working, as expected. 
As this is based upon the context of Watchtower's configuration, this includes scope-based filtering, as shown below.

Test Configuration:

```yaml
services:
  watchtower-test:
    container_name: watchtower-test
    image: watchtower:test
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - WATCHTOWER_DEBUG=true
      - WATCHTOWER_SCOPE=watchtower-test
      - WATCHTOWER_HTTP_API_TOKEN=test
      - WATCHTOWER_HTTP_API_CONTAINERS=true
    networks:
      - watchtower-test
    ports:
      - 8080:8080
    restart: unless-stopped
    labels:
      - com.centurylinklabs.watchtower.scope=watchtower-test
networks:
  watchtower-test:
    name: watchtower-test
```

Request:

```bash
curl -X GET -H "Authorization: Bearer test" localhost:8080/v1/containers
```

Response:

```json
{
    "api_version": "v1",
    "containers": [
        {
            "name": "watchtower-test",
            "image": "watchtower:test",
            "image_id": "sha256:9053a97e02420c1a617c05561aeb84232f825553831e149d341752cf1796c117",
            "digest": "sha256:9053a97e02420c1a617c05561aeb84232f825553831e149d341752cf1796c117"
        }
    ],
    "count": 1,
    "timestamp": "2026-05-29T18:51:55Z"
}
```

I am accepting this PR, because the addition of this endpoint is consistent with Watchtower's purpose: Docker container lifecycle management (i.e. container updates). With that said, I do not recommend using Watchtower in a commercial or production environment.

From what you described, I would recommend considering a solution that is specifically tailored to your use case and needs. If you have an interest in working on a such a project, then please feel free to let me know.

**Review by nicholas-fedor — APPROVED** [review]

**Comment by nicholas-fedor** [comment]
@all-contributors add @barnabasbusa for code and documentation
